tests/test_utils.py

```diff
@@ -42,3 +45,107 @@ def test_numpy_round_trip(batch_size):
     np.testing.assert_allclose(
         R_np, R_recon, rtol=1e-5, atol=1e-6, err_msg=f"NumPy round-trip failed for batch size {batch_size}"
     )
+
+
+@pytest.mark.parametrize("batch_size", [0, 1, 100])
+@pytest.mark.parametrize("func", ["T", "R"])
+def test_torch_identity_transform(batch_size, func):
+    if batch_size == 0:
+        pos = torch.randn(3)
+        b_pos = torch.randn(10, 3)
+    else:
+        pos = torch.randn(batch_size, 3)
+        b_pos = torch.randn(batch_size, 10, 3)
+
+    if func == "R":
+        identity = torch.eye(3)
+        batched_identity = torch.eye(3).reshape(1, 3, 3).repeat(batch_size, 1, 1)
+        transform = transform_by_R
+    elif func == "T":
+        identity = torch.eye(4)
+        batched_identity = torch.eye(4).reshape(1, 4, 4).repeat(batch_size, 1, 1)
+        transform = transform_by_T
+
+    transformed = transform(pos, identity).cpu().numpy()
+    np.testing.assert_allclose(
+        pos.cpu().numpy(),
+        transformed,
+        rtol=1e-5,
+        atol=1e-6,
+        err_msg=f"NumPy identity_transform failed for batch size {batch_size}",
+    )
+    transformed = transform(b_pos, identity).cpu().numpy()
+    np.testing.assert_allclose(
+        b_pos.cpu().numpy(),
+        transformed,
+        rtol=1e-5,
```

Comment:
Why this accuracy?

I should probably disable auto-start of genesis for every single tests. There is no point here.